v2/samples/apimanagement/v1api20220801/v1api20220801_productapi.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: apimanagement.azure.com/v1api20220801
+kind: ProductApi
+metadata:
+  name: sampleproductapi
+  namespace: default
+spec:
+  azureName: sampleapi
+  owner:
+    name: sampleapimproduct
```

Comment:
We're excluding samples test for APIM/Product [here](https://github.com/Azure/azure-service-operator/blob/main/v2/internal/testcommon/samples_tester.go#L64), hence why the samples test would be failing. Since you've done a fix for deleteSubscription in #3467, should be alright to include the sample back in and record.

Reply:
I think further investigation is needed as to why these samples don't run in the tests. Product has little use without ProductApi and ProductPolicy, but I would like to re-enable these tests when v2 of APIM is ready as this will cut the tests down from +1.5 hours to a few minutes.

For teh time being we are now not running Sample Tests for Product/Subscription/ProductPolicy/ProductApi.

However, if we get this PR, @ansulgoenka can uncomment these tests

What say you, @super-harsh ? :-)